api/src/app/modules/project/project.service.ts

```diff
@@ -669,203 +934,365 @@ export class ProjectService extends DataService<Project, Repository<Project>, Pr
+      );
     }
 
-    if (isNil(request.commentingClosedDate || !dayjs(request.commentingClosedDate, this.DATE_FORMAT).isValid())) {
-      throw new BadRequestException("Missing valid Commenting Closed Date");
+    if (
+      isNil(
+        request.commentingClosedDate ||
+          !dayjs(request.commentingClosedDate, this.DATE_FORMAT).isValid()
+      )
+    ) {
+      throw new BadRequestException('Missing valid Commenting Closed Date');
     }
 
-    const dayDiff = DateTimeUtil.diffNow(request.commentingClosedDate, DateTimeUtil.TIMEZONE_VANCOUVER, 'day');
+    const dayDiff = DateTimeUtil.diffNow(
+      request.commentingClosedDate,
+      DateTimeUtil.TIMEZONE_VANCOUVER,
+      'day'
+    );
     if (dayDiff < 1) {
       throw new BadRequestException(`Unable to update Commenting Closed Date to ${request.commentingClosedDate}.
         The earliest new date allowed is tomorrow.`);
     }
 
     const workflowStateCode = entity.workflowStateCode;
-    if (WorkflowStateEnum.INITIAL != workflowStateCode && 
-      WorkflowStateEnum.COMMENT_OPEN != workflowStateCode && 
-      WorkflowStateEnum.PUBLISHED != workflowStateCode) {
-      throw new BadRequestException("Can only change Commenting Closed Date at Initial, Published, or Commenting Open status.");
+    if (
+      WorkflowStateEnum.INITIAL != workflowStateCode &&
+      WorkflowStateEnum.COMMENT_OPEN != workflowStateCode &&
+      WorkflowStateEnum.PUBLISHED != workflowStateCode
+    ) {
+      throw new BadRequestException(
+        'Can only change Commenting Closed Date at Initial, Published, or Commenting Open status.'
+      );
     }
 
     // doing update.
-    const updateCount = (await this.repository.update(projectId, 
-                            { revisionCount: entity.revisionCount + 1, 
-                              updateUser: user.userName,
-                              updateTimestamp: new Date(),
-                              commentingClosedDate: request.commentingClosedDate
-                            }
-                          )).affected;
+    const updateCount = (
+      await this.repository.update(projectId, {
+        revisionCount: entity.revisionCount + 1,
+        updateUser: user.userName,
+        updateTimestamp: new Date(),
+        commentingClosedDate: request.commentingClosedDate,
+      })
+    ).affected;
     if (updateCount != 1) {
-      throw new InternalServerErrorException("Error updating object");
+      throw new InternalServerErrorException('Error updating object');
     }
 
     return true;
   }
 
+  /**
+   * Returns the number of non-initial published FOM projects. 
+   * Used by analytics dashboard module.
+   * 
+   * @param startDate - The start of the date range (inclusive, YYYY-MM-DD)
+   * @param endDate - The end of the date range (inclusive, YYYY-MM-DD)
+   * @returns Promise resolving to the count of projects matching the criteria
+   */
+  async getNonInitialPublishedProjectCount(
```

Comment:
I applied the formatter on this file, the real changes start from here.

Reply:
I can revert it back, but I also feel long term we should format the code, but I'm okay to revert it back in this pr.